src/tscore/unit_tests/test_Extendible.cc

```diff
@@ -209,20 +209,36 @@ TEST_CASE("Extendible", "")
     CHECK(testField::alive == 0);
   }
 
+  INFO("AcidPtr AcidCommitPtr malloc ptr int");
+  {
+    void *mem            = malloc(sizeof(AcidPtr<int>));
+    AcidPtr<int> &reader = *(new (mem) AcidPtr<int>);
```

Comment:
I don't see what it adds to put the AcidPtr instance in the heap, why not just:

`AcidPtr<int> reader;`

Avoiding the need for  the explicit destructor call and malloc/free calls.  Why doesn't AcidPtr<T> use perfect forwarding to that constructors of T with parameters can be used?

I suggest adding here:
`std::shared_ptr<T> sp = reader.getPtr();`

Reply:
I asked @a-canary about this and he pointed out this is closer to how it will be used in practice. That is, the schema asks for a chunk of memory embedded in a larger chunk and that must be initialized in place.

Reply:
Doesn't seem relevant, but harmless worst-case.  Nit-pick, it's a good habit to use sentries:  https://godbolt.org/z/_hh6WT

Reply:
yes, test_AcidPtr.cc contians a simple test like that. 
This is mimicking how it behaves in a block allocation like Extendible.